src/Compilers/CSharp/Portable/BoundTree/DecisionTreeBuilder.cs

```diff
@@ -26,17 +26,63 @@ internal abstract class DecisionTreeBuilder
         protected readonly Symbol _enclosingSymbol;
         protected readonly Conversions _conversions;
         protected HashSet<DiagnosticInfo> _useSiteDiagnostics = new HashSet<DiagnosticInfo>();
+        private Dictionary<TypeSymbol, LocalSymbol> localByType = new Dictionary<TypeSymbol, LocalSymbol>();
```

Comment:
>new Dictionary<TypeSymbol, LocalSymbol>(); [](start = 66, length = 42)

Should this dictionary be sensitive to dynamic/object, tuple names and custom modifier differences?